tests/stdlib/tordsets.nim

```diff
@@ -0,0 +1,97 @@
+import std/ordsets
+import std/sets
+
+from sequtils import toSeq
+from algorithm import sorted
+
+proc sortedPairs[T](t: T): auto = toSeq(t.pairs).sorted
+template sortedItems(t: untyped): untyped = sorted(toSeq(t))
+
+type id = distinct int
+proc cmp(a: id, b: id): int {.borrow.}
+proc `==`(a: id, b: id): bool {.borrow.}
+proc `<`(a: id, b: id): bool {.borrow.}
+
+block: # we use HashSet as groundtruth, it's well tested elsewhere
+  template testDel(A: typedesc, t: typed, t0: typed) =
+
+    block:
+      template checkEquals() =
+        doAssert t.len == t0.len
+        for k in t0:
+          doAssert k in t
+        for k in t:
+          doAssert k in t0
+
+        doAssert sortedItems(t) == sortedItems(t0)
+
+      template incl2(i) =
+        t.incl i
+        t0.incl i
+
+      template excl2(i) =
+        t.excl i
+        t0.excl i
+
+      var expected: seq[`A`]
+      let n = 100
+      let n2 = n*2
+      for i in 0..<n:
+        incl2(`A`(i))
```

Comment:
nit: why the quoting of A?
ditto elsewhere

Reply:
It's a typedesc arg for the template, isn't the quoting required for the typecast?

Reply:
no, this works:
```nim
  template fn(A: typedesc) = echo A(1)
  fn(float)
```